services/volume/src/service.rs

```diff
@@ -0,0 +1,333 @@
+#![allow(clippy::unit_arg)]
+
+use super::*;
+use common::wrapper::v0::*;
+
+/// Volume service implementation methods
+#[derive(Clone, Debug, Default)]
+pub(super) struct VolumeSvc {
+    registry: Registry<NodeWrapperVolume>,
+}
+
+impl VolumeSvc {
+    /// New Service with the update `period`
+    pub fn new(period: std::time::Duration) -> Self {
+        let obj = Self {
+            registry: Registry::new(period),
+        };
+        obj.start();
+        obj
+    }
+    /// Start registry poller
+    fn start(&self) {
+        self.registry.start();
+    }
+
+    /// Get all pools from node or from all nodes
+    async fn get_node_nexuses(
+        &self,
+        node_id: Option<String>,
+    ) -> Result<Vec<Nexus>, SvcError> {
+        Ok(match node_id {
+            None => self.registry.list_nexuses().await,
+            Some(node_id) => self.registry.list_node_nexuses(&node_id).await,
+        })
+    }
+
+    /// Get nexuses according to the filter
+    #[tracing::instrument(level = "debug", err)]
+    pub(super) async fn get_nexuses(
+        &self,
+        request: &GetNexuses,
+    ) -> Result<Nexuses, SvcError> {
+        let filter = request.filter.clone();
+        let nexuses = match filter {
+            Filter::All => self.get_node_nexuses(None).await?,
```

Comment:
Filter all or filter None?

Reply:
I wonder if Filter is wrong here. We're not filtering, it's more like querying so maybe it should be Query?